alica_test_utility/src/TestContext.cpp

```diff
@@ -62,4 +64,235 @@ bool TestContext::isPlanActive(int64_t id) const
+    if (rp != nullptr) {
+        return rp->isBehaviour() ? rp->getBasicBehaviour() : nullptr;
+    }
+    return nullptr;
+}
+
+BasicPlan* TestContext::getActivePlan(const std::string& name)
+{
+    auto rp = getRunningPlan(name);
+    if (rp != nullptr) {
+        return rp->isBehaviour() ? nullptr : rp->getBasicPlan();
+    }
+    return nullptr;
+}
+
+bool TestContext::setTransitionCond(const std::string& runningPlanName, const std::string& inState, const std::string& outState)
+{
+    return setTransitionCond(runningPlanName, inState, outState, true);
+}
+
+bool TestContext::resetTransitionCond(const std::string& runningPlanName, const std::string& inState, const std::string& outState)
+{
+    return setTransitionCond(runningPlanName, inState, outState, false);
+}
+
+bool TestContext::resetAllTransitions(const std::string& runningPlanName)
+{
+    auto* rp = getRunningPlan(runningPlanName);
+    if (!rp) {
+        return false;
+    }
+    return resetAllTransitions(rp);
+}
+
+bool TestContext::resetAllTransitions(RunningPlan* rp)
+{
+    const auto* plan = rp->getActivePlanAsPlan();
+    if (!plan) {
+        return false;
+    }
+    for (const auto* transition : plan->getTransitions()) {
+        const auto& inState = transition->getInState()->getName();
+        const auto& outState = transition->getOutState()->getName();
+        setTransitionCond(rp, inState, outState, false);
+    }
+    return true;
+}
+
+bool TestContext::isSuccess(const BasicBehaviour* beh) const
+{
+    return beh->isSuccess();
+}
+
+bool TestContext::isSuccess(const BasicPlan* plan) const
+{
+    return plan->getPlanContext()->getActiveState()->isSuccessState();
+}
+
+bool TestContext::isStateActive(const std::string& runningPlanName, const std::string& stateName)
+{
+    auto* rp = getRunningPlan(runningPlanName);
+    if (!rp) {
+        return false;
+    }
+    return rp->getActiveState() ? (rp->getActiveState()->getName() == stateName) : false;
+}
+
+RunningPlan* TestContext::getRunningPlan(const std::string& name)
+{
+    if (name.empty()) {
+        return nullptr;
+    }
+    return name[0] == '/' ? /* fully qualified name */ followRunningPlanPath(name) : /* just name of beh/plan */ searchRunningPlanTree(name);
+}
+
+RunningPlan* TestContext::searchRunningPlanTree(const std::string& name)
+{
+    if (!_engine->getPlanBase().getRootNode()) {
+        return nullptr;
+    }
+    std::vector<RunningPlan*> results;
+    std::queue<RunningPlan*> q;
+    q.push(_engine->getPlanBase().getRootNode());
+    while (!q.empty()) {
+        auto cur = q.front();
+        q.pop();
+        if (!cur->isActive()) {
+            return nullptr;
+        }
+        if (getRunningPlanName(cur) == name) {
+            results.push_back(cur);
+        }
+        for (auto child : cur->getChildren()) {
+            q.push(child);
+        }
+    }
+    if (results.empty() || results.size() > 1) {
+        return nullptr;
+    }
+    return results.front();
```

Comment:
Can we add warnings if we cant find the correct node / find multiple with the same name? This could help with debugging failing tests

Reply:
can probably added a `getFailureInfo` method like the lbc mockups which returns the previous failure